api/api.go

```diff
@@ -599,7 +599,7 @@ func NewClient(config *Config) (*Client, error) {
 	if len(parts) == 2 {
 		switch parts[0] {
 		case "http":
-			config.Scheme = "http"
+			// Never revert to http if TLS was explicitly requested.
```

Comment:
This looks like it was introduced in 12216583a124e25de630fb8d3e96ed0c7b431142 which is a shockingly large change, and I have to assume was not intended. Disabling TLS based on the URL is not safe.